Make bundled theme generic (pre checkout)

renatho opened the pull request:
<!--
Link a related issue to this PR. If the PR does not immediately resolve the issue,
for example, it requires a separate deployment to production, avoid
using the "fixes" keyword and instead attach the [Status] Fix Inbound label to
the linked issue.
-->

Related to https://github.com/Automattic/themes/issues/7498

## Proposed Changes

* It's a refactor to make the bundle theme more generic, so we can easily add other themes on the future. Basically, all the settings are part of the new `client/my-sites/theme/bundle-settings.tsx`.
  * This doesn't include the installation flow yet. I decided to explore it as a separate thing and raise another PR to avoid more complexity in this one.
* ~~I also added the "[Status] String Freeze" label to this PR, so we make sure we don't break the translations of strings that are already in the site but had their structure changed.~~ https://github.com/Automattic/wp-calypso/pull/84483#issuecomment-1828504748

## Testing Instructions

<!--
Add as many details as possible to help others reproduce the issue and test the fix.
"Before / After" screenshots can also be very helpful when the change is visual.
-->

Themes

* Navigate to http://calypso.localhost:3000/themes
* Select a site.
* Search for "Tsubaki".
* Make sure it continues working as properly, as previously.

Design Picker and checkout

* Navigate to the http://calypso.localhost:3000/start
* Navigate through the flow until get to the Design Picker.
* Filter by store.
* Check the Tsubaki theme and make sure it continues working properly, as previously.
* Choose the theme, click on "Upgrade Plan", and make sure it goes to the checkout to buy the business plan.
* Complete the checkout, and check that it returns to the theme and that you can continue.

## Screenshots

**Themes**

<img width="709" alt="Screenshot 2023-11-27 at 16 06 43" src="https://github.com/Automattic/wp-calypso/assets/876340/dfee8608-4da1-4acb-afec-a485378753f9">
<img width="803" alt="Screenshot 2023-11-27 at 16 06 59" src="https://github.com/Automattic/wp-calypso/assets/876340/4b2a7cf6-bf7a-4042-a764-c28b0c135fe6">

**Design Picker**

<img width="1025" alt="Screenshot 2023-11-27 at 16 05 56" src="https://github.com/Automattic/wp-calypso/assets/876340/1863223b-2371-4e75-8fa4-9975f98542d5">
<img width="863" alt="Screenshot 2023-11-27 at 16 06 12" src="https://github.com/Automattic/wp-calypso/assets/876340/56864fe5-ffdf-4286-a4c9-edfcb923ea02">

## Pre-merge Checklist

<!--
Complete applicable items on this checklist **before** merging into trunk. Inapplicable items can be left unchecked.

Both the PR author and reviewer are responsible for ensuring the checklist is completed.
-->

- [x] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [x] Have you checked for TypeScript, React or other console errors?
- [x] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [x] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

About the first item, I didn't test exhaustively the browser support because it shouldn't change anything for the browser.

renatho committed:
Refactor WooCommerceBundledBadge component to be generic receiving the details through props

renatho committed:
Update Woo icon to simplify styles

renatho committed:
Refactor sass code

renatho committed:
Add container for the icon

renatho committed:
Move WooCommerce badge settings to a settings file

renatho commented on client/my-sites/theme/bundle-settings.jsx:
```diff
@@ -0,0 +1,17 @@
+	[ WOOCOMMERCE_THEME ]: {
+		name: 'WooCommerce',
+		badgeIcon: (
+			<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24">
```
The original icon was changed to this one, so it uses the `currentColor` to set the icon as white, and the "Woo" text is transparent. It will make it easier to create the icons generically for all bundles.

renatho committed:
Add selector to get theme software set

renatho committed:
Make WooCommerce theme bundle generic, in order to allow other bundles

renatho commented on packages/design-picker/src/constants.ts:
```diff
@@ -80,5 +80,5 @@ export const DEFAULT_ASSEMBLER_DESIGN = {
 export const PREMIUM_THEME = 'premium';
 export const DOT_ORG_THEME = 'dot-org';
-export const WOOCOMMERCE_THEME = 'woocommerce';
+export const BUNDLED_THEME = 'bundled';
```
Is it a good name?

daledupreez replied:
Good question. I'd consider flipping it around so we're looking at `THEME_WITH_BUNDLED_SOFTWARE`, but that looks nothing like the other constants. 🤔

renatho replied:
Yeah! 🤔
I'll keep this one for now, but if any idea comes to your mind, feel free to let me know that I can change it. 😉

renatho committed:
Return slugs instead of names

renatho commented on client/state/themes/selectors/get-theme-software-set.js:
```diff
@@ -0,0 +1,34 @@
+			return [];
+		}
+
+		return themeSoftwareSetTaxonomy.map( ( item ) => item.slug );
```
Is it the `slug` the correct reference we should use here? I went with this approach based on this interface I found in the code during the refactor: https://github.com/Automattic/wp-calypso/blob/230e74383a1756da94e1a65a3c0401411c1b39e8/packages/design-picker/src/types.ts#L73

renatho commented on client/my-sites/theme/bundle-settings.tsx:
```diff
@@ -0,0 +1,25 @@
+import type { ReactElement } from 'react';
```
Is it the best place to add this file with bundle settings?

renatho committed:
Refactor occurrences of is_bundled_with_woo_commerce to make it generic for any bundle

renatho committed:
Remove TODOs since it will have multiple occurrences

It will be documented and discussed in the PR.

renatho committed:
Refactor messages to be generic using bundle settings

renatho commented on client/components/theme-type-badge/index.tsx:
```diff
@@ -44,6 +46,8 @@ const ThemeTypeBadge = ( {
 		} );
 	}, [ type, themeId, isLockedStyleVariation ] );
 
+	const themeSoftware = themeSoftwareSet[ 0 ];
```
In some places, we are hard coding the `themeSoftwareSet[ 0 ]` to get the first one. Do we plan to have multiple plugins bundled with a team? Sensei LMS + WooCommerce, for example. If so, we need some more refactor.

dsas replied:
You can have multiple plugins already - the Woo software set includes Woo and Woo Payments plugins. 

I think the question is more 'do we plan to have multiple software sets for a theme'. There aren't any plans that I know of, and from a data pov it could just be a new software set that contains e.g. woo+sensei plugins, it's from a UI perspective it gets trickier - which badge do we display etc.

I don't think we should worry too much about this right now, lets assume that there are always zero or one software sets for any given theme.

daledupreez replied:
For this thread, it may be worth considering a way to identify the _primary_ software set: what does the user think they are getting/installing when they pick this theme?

renatho replied:
> I think the question is more 'do we plan to have multiple software sets for a theme'.

Sorry! That's correct!

> I think the question is more 'do we plan to have multiple software sets for a theme'.

Could you elaborate a little more your idea @daledupreez? I'm not sure if I understand it.
Do you mean improving the UI with the details of what would be installed? Or something in the showcase editor to set the primary software set? Or something else not related to this at all? 🤭

renatho commented on client/my-sites/theme/bundle-settings.tsx:
```diff
@@ -0,0 +1,25 @@
+	'woo-on-plans': {
+		name: 'WooCommerce',
+		badgeIcon: (
+			<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24">
```
The original icon was changed to this one, so it uses the `currentColor` to set the icon as white, and the "Woo" text is transparent. It will make it easier to create the icons generically for all bundles.

renatho committed:
Fix missing variable

renatho committed:
Update badge icon to a component

renatho committed:
Remove todo comment

The reference for this name appears in many places, so I'm keeping at for now to avoid any accidental issue.

renatho commented on client/my-sites/theme/bundle-settings.tsx:
```diff
@@ -0,0 +1,22 @@
+}
+
+const bundleSettings: BundleSettings = {
+	'woo-on-plans': {
```
I was planning to change the `'woo-on-plans'` to a constant. But it appears in many places, so I didn't want to risk an accidental issue for now.

renatho commented on client/components/theme-type-badge/tooltip.tsx:
```diff
@@ -163,11 +178,20 @@ const ThemeTypeBadgeTooltip = ( {
+			const bundleName = bundleSettings[ themeSoftware ].name;
+
+			if ( isIncludedCurrentPlan ) {
+				message = translate( 'This %(bundleName)s theme is included in your plan.', {
```
Currently, for many of the texts, I'm just interpolating the bundle name. If we want a specific text for any part, we can add it to the `bundleSettings`.

renatho committed:
Fix design picker badge

renatho committed:
Add a separate text for upsell description

renatho commented on client/my-sites/theme/bundle-settings.tsx:
```diff
@@ -0,0 +1,28 @@
+
+interface BundleSettings {
+	[ key: string ]: {
+		name: string;
```
The name is used to represent the bundle name. Currently, we have only "WooCommerce", but it doesn't necessarily need to be a plugin name. It could be "Special", for example.

This is used as a label for the bundle badge, and also to concat some strings, mostly used as "WooCommerce theme", or "Special theme" (in a future case).

dsas replied:
Is it worth a comment to that effect? It's just a text label.

renatho replied:
Do you mean something like this? https://github.com/Automattic/wp-calypso/pull/84483/commits/de1020aacb31f5baa2dfaf532bf696ae780694f8

renatho commented on client/my-sites/theme/bundle-settings.tsx:
```diff
@@ -0,0 +1,28 @@
+import { translate } from 'i18n-calypso';
```
Is it the best place to add this file with bundle settings?

dsas commented on client/my-sites/theme/bundle-settings.tsx:
```diff
@@ -0,0 +1,32 @@
+interface BundleSettings {
+	[ key: string ]: {
+		name: string;
+		BadgeIcon: React.ComponentType;
```
minor nitpick: the casing of `BadgeIcon` doesn't match the other BundleSettings properties.

renatho replied:
I had used that on purpose because it's a React component. But I think it's fine to keep the lowercase here since it's the object and we don't use it directly as the component.

https://github.com/Automattic/wp-calypso/pull/84483/commits/a93cd3e8e1ebac1b30aa6bb07393ab78ca6979cf

dsas commented on packages/components/src/theme-type-badge/bundled-badge/style.scss:
```diff
@@ -1,4 +1,4 @@
-.woocommerce-bundled-badge {
+.bundled-badge {
 	--woocommerce-purple-50: #7f54b3;
```
this is woo specific so will need changing at some point 🤔

dsas replied:
Same thing applies to line 13 below

renatho replied:
Good catch! It's supposed to be generic here now: https://github.com/Automattic/wp-calypso/pull/84483/commits/078d3465c224b5fc5cd402205c77a2a422a112f8

dsas reviewed (commented):
I've just noticed that something else will need changing (I didn't notice when I was documenting, sorry). The [ThemeUpgradeModal](https://github.com/Automattic/wp-calypso//blob/a4302dc9129c75d7b99fee7e34eac8cb134c09ee/client/components/theme-upgrade-modal/index.tsx#L132) has a Woo logo and text hard-coded. 

You can see the modal by going /start, working through the process until theme selection, then choose a Woo theme and press 'Unlock theme'.

It's an additional thing though rather than this PR breaking something. I'm sure more additional changes will come to light as the Sensei bundle is defined and worked on.

A small thing I've noticed is that the badge has subtly changed, it's about 5px narrower in the PR. See:
https://github.com/Automattic/wp-calypso/assets/93301/bc15509e-cbbc-4232-8936-b532a0b6d984

renatho committed:
Add setting for bundled badge color

renatho committed:
Update property name to match other properties casing

renatho committed:
Move variables to the scope where it's used

renatho committed:
Update icon and color settings to be generic

daledupreez commented on client/components/theme-type-badge/tooltip.tsx:
```diff
@@ -163,11 +178,20 @@ const ThemeTypeBadgeTooltip = ( {
+			const bundleName = bundleSettings[ themeSoftware ].name;
+
+			if ( isIncludedCurrentPlan ) {
+				message = translate( 'This %(bundleName)s theme is included in your plan.', {
+					args: { bundleName },
+				} );
+			} else {
+				message = createInterpolateElement(
+					translate( 'This %(bundleName)s theme is included in the <Link>Business plan</Link>.', {
+						args: { bundleName },
+						textOnly: true,
+					} ),
```
We should add some translation comments to make it clearer what values we expect to pass in for `bundleName`.

renatho replied:
Sure! Added in https://github.com/Automattic/wp-calypso/pull/84483/commits/707dffc27cde6ca5852c39af4746d6466bd58a78

daledupreez commented on client/my-sites/theme/bundle-settings.tsx:
```diff
@@ -0,0 +1,32 @@
+	};
+}
+
+const bundleSettings: BundleSettings = {
```
Have you considered wrapping this in a hook? I am asking because the current direct lookups in `bundleSettings[ type ]` feel a bit awkward _and_ because the use of `translate` directly may miss contextual information that's needed to get the right locale etc. If we're in a hook, we can switch to `const translate = useTranslate();` within the hook.

renatho replied:
Gotcha! I didn't consider this scenario where the `translate` could change.
Changed it here: https://github.com/Automattic/wp-calypso/pull/84483/commits/76d49f2aa201ceb84ff9317478fac4f334bc6997
And then simplified a bit more here: https://github.com/Automattic/wp-calypso/pull/84483/commits/ab8af367e962efadaac36b1d6ce7095337a801e6

daledupreez commented on client/my-sites/theme/main.jsx:
```diff
@@ -1016,14 +1018,25 @@ class ThemeSheet extends Component {
+
+			const bundleName = bundleSettings[ themeSoftware ].name;
+
+			return translate( 'Access this %(bundleName)s theme with a Business plan!', {
+				args: { bundleName },
+			} );
```
Please add a translators comment to make it easier to translate this effectively.

renatho replied:
Sure! Added in https://github.com/Automattic/wp-calypso/pull/84483/commits/707dffc27cde6ca5852c39af4746d6466bd58a78

daledupreez commented on client/state/themes/selectors/get-theme-software-set.js:
```diff
@@ -0,0 +1,34 @@
+ * @param {string} themeId Theme ID
+ * @returns {string[]} Array with the name of the softwares.
+ */
+export const getThemeSoftwareSet = createSelector(
```
Might it be worth writing this code in TypeScript given that it's net new?

renatho replied:
Sure and noted!
Updated in https://github.com/Automattic/wp-calypso/pull/84483/commits/fb56485238e3f67b125ff4f4aaa27a6913798ad3

daledupreez commented on packages/components/src/theme-type-badge/bundled-badge/index.tsx:
```diff
@@ -0,0 +1,91 @@
+import classNames from 'classnames';
+import { FunctionComponent, useRef, useMemo, useState } from 'react';
```
Nit: can you import only the `FunctionComponent` type?
```suggestion
import { type FunctionComponent, useRef, useMemo, useState } from 'react';
```

daledupreez commented on packages/components/src/theme-type-badge/bundled-badge/index.tsx:
```diff
@@ -0,0 +1,91 @@
+				}
+			},
+		};
+	}, [ isClickable, isPopoverVisible ] );
```
Shouldn't the dependencies include `setIsPressed` and `setIsPopoverVisible`?

renatho replied:
In this case, it's not needed because they are functions from the `useState` that never change.

daledupreez commented on packages/components/src/theme-type-badge/bundled-badge/style.scss:
```diff
@@ -21,55 +21,54 @@
 	}
 
-	svg {
+	&__icon {
```
Nit: can you avoid this shorthand? While it's less text, it will make finding the relevant CSS much harder using normal find/search operations. This applies here and below.

renatho replied:
Yep! Updated in https://github.com/Automattic/wp-calypso/pull/84483/commits/b492dc61666185a3e4e8fcd666f7745deb20f48f. Could you confirm if it matches the standards or if I should change anything else?

Basically, the commit removes the cases where it was concating class names. It keeps some `&` for multiple selectors in the same element and the nested structure though.

daledupreez reviewed (commented):
I haven't tested the changes, but they look relatively reasonable overall.

renatho committed:
Make bundle modal generic

renatho committed:
Decrease padding around svg

With this the image matches a little better the current state of the image in the modal

renatho commented:
> I've just noticed that something else will need changing (I didn't notice when I was documenting, sorry). The [ThemeUpgradeModal](https://github.com/Automattic/wp-calypso//blob/a4302dc9129c75d7b99fee7e34eac8cb134c09ee/client/components/theme-upgrade-modal/index.tsx#L132) has a Woo logo and text hard-coded.

Thank you for checking and also catching that! I updated it here: https://github.com/Automattic/wp-calypso/pull/84483/commits/c97bc3168facd873c71a3e90977fe4df857bac3e

> A small thing I've noticed is that the badge has subtly changed, it's about 5px narrower in the PR. See:
https://github.com/Automattic/wp-calypso/assets/93301/bc15509e-cbbc-4232-8936-b532a0b6d984

The reason is because of a probable mistake. Previously, the SVG had `width="24" height="24"` on the tag and a CSS style with `height: 20px`. I didn't bring the `width="24" and height="24"` to the new icon, but I set it as `20px` height, automatically setting the width to `20px`. We could make it more similar to the previous state by adding a CSS with `width: 24px; height: 20px;`, but I think it's better to work as a square since we're making it generic now. WDYT?

After your review, I did just another small tweak on the SVG to decrease the transparent are around to use the same one in the modal.

Adding some screenshots of the previous state and now to see what you think:

**Before:**
<img width="149" alt="Screenshot 2023-11-28 at 19 04 13" src="https://github.com/Automattic/wp-calypso/assets/876340/52d22f13-3e90-4071-b232-4ac9fb047bbd">

**Now:**
<img width="145" alt="Screenshot 2023-11-28 at 19 04 02" src="https://github.com/Automattic/wp-calypso/assets/876340/e97744e0-941d-407a-a952-433f9b94b998">

**Before:**
<img width="790" alt="Screenshot 2023-11-28 at 19 05 13" src="https://github.com/Automattic/wp-calypso/assets/876340/0212f509-57ce-4c60-b898-70346b7f7a01">

Notice that it also had a spacing after the `img`, but the reason is a [common issue](https://www.geeksforgeeks.org/how-to-get-rid-of-the-gap-under-the-image/) that happen with `img` tag because it's inline. Changing to `svg` the issue doesn't happen so we have a small difference on that. We could increase the spacing to match the previous state, but I think it was an unexpected issue.

**Now:**
<img width="789" alt="Screenshot 2023-11-28 at 19 05 01" src="https://github.com/Automattic/wp-calypso/assets/876340/d9607d07-af40-4942-b452-bd06157533f2">

renatho committed:
Add comment to document field

renatho committed:
Merge branch 'trunk' of github.com:Automattic/wp-calypso into change/make-bundled-theme-generic

dsas commented on client/state/themes/selectors/can-use-theme.js:
```diff
@@ -38,7 +38,7 @@ export function canUseTheme( state, siteId, themeId ) {
+	if ( type === BUNDLED_THEME ) {
 		return (
 			siteHasFeature( state, siteId, WPCOM_FEATURES_PREMIUM_THEMES ) &&
 			siteHasFeature( state, siteId, FEATURE_WOOP ) &&
```
Checking for FEATURE_WOOP won't be technically correct for sensei 🤔 (though will work fine) It means "does this plan support Woo" and so currently resolves to business, e-commerce and pro plans, i.e. atomic capable ones.

The same check is made lower down for marketplace themes already and doesn't make sense there either 🤔

dsas replied:
As discussed on slack, we've removed it from the marketplace theme and here the code should check it depending on whether the software set is woo-on-plans or not.

p1701355655069369/1700605581.879429-slack-C048CUFRGFQ

renatho replied:
The check from the marketplace was removed in https://github.com/Automattic/wp-calypso/pull/84483/commits/6cf6789e2f0084cac58830e471f6ab8bd0bf3f26.

And some checks are done based on the software set now https://github.com/Automattic/wp-calypso/pull/84483/commits/d314aab14847a6336a59a8d2f4c1206bd662faf2. Notice that I'm adding this logic to the `canUseTheme` instead of the bundle settings because it was the simplest way I found to implement it now, and from my understanding I think it is an exception for Woo. If it starts getting common for other softwares, we could return with an approach like https://github.com/Automattic/wp-calypso/pull/84483/commits/ab8af367e962efadaac36b1d6ce7095337a801e6, which has a static part of the settings, so we could export it and use in the selector `canUseTheme`.

renatho committed:
Merge branch 'trunk' of github.com:Automattic/wp-calypso into change/make-bundled-theme-generic

renatho committed:
Add translator comments

renatho committed:
Transform new selector to typescript

renatho committed:
Import only FunctionComponent type

Co-authored-by: daledupreez <dale@automattic.com>

renatho committed:
Update sass file to not use name concating

The purpose is to be easier to find the classes in a search operation.

renatho committed:
Move bundle settings to a hook

So it can re-render when translations change.

renatho commented on client/landing/stepper/declarative-flow/internals/steps-repository/design-setup/design-picker-design-title.tsx:
```diff
@@ -29,11 +30,21 @@ const DesignPickerDesignTitle: FC< Props > = ( { designTitle, selectedDesign } )
 	);
 
-	const showBundledBadge = selectedDesign.is_bundled_with_woo_commerce;
+	const bundleSettings = useBundleSettings( selectedDesign.slug );
```
If I understand correctly, the `selectedDesign` always will be a theme, so `selectedDesign.slug` is the same thing as `themeId` from other parts of the application. Is it correct?

dsas replied:
Yes, that is true now, we removed the last feature where this wasn't the case in June (D108448-code)

renatho committed:
Remove separation of the static and dynamic part

The benefit isn't worth the extra complexity.

renatho committed:
Improve vertical alignment of the icon

okmttdhr commented on packages/data-stores/src/starter-designs-queries/use-starter-designs-query.ts:
```diff
@@ -95,10 +95,6 @@ function apiStarterDesignsToDesign( design: StarterDesign ): Design {
 
 	const is_externally_managed = design.theme_type === 'managed-external';
 
-	const is_bundled_with_woo_commerce = ( design.software_sets || [] ).some(
-		( { slug } ) => slug === 'woo-on-plans'
-	);
```
Hello, @renatho! I have a question 🙋‍♀️ 
The original function `is_bundled_with_woo_commerce` checks for the presence of `woo-on-plans`. However, the PR modifies this to check if `software_sets.length > 0`. 
As far as checking `useBundleSettings`, there will be additional types of `software_sets`. In that case, these changes will break the original behavior in the future, am I correct?

renatho replied:
Hi @okmttdhr! Correct! That's the idea.
We want to consider the theme as a bundled theme when it has any software set. And apply the specific settings for each software set.
Currently, it has only the `woo-on-plans`. But in a following PR, we also want to introduce a software set for Sensei LMS, which will have its specific settings.

okmttdhr replied:
Thanks for your comment, @renatho! 
This PR changes that instead of checking for `woo-on-plans`, we should only verify if `software_sets.length > 0` (as seen in `design-setup/unified-design-picker.tsx` for example). This approach means that the condition will be `true` if any new software sets introduced in the future not limited to Woo. 

And I assume this is the intended purpose of this PR. I'm highlighting this to clarify my understanding 🙂

renatho replied:
That's completely correct! 🙂
For context, there are some more details in pekYwv-3rN-p2 and in p1700605581879429-slack-C048CUFRGFQ.

renatho committed:
Remove not used check

dsas reviewed (commented):
Doing some testing and after purchasing a business plan via /themes I *don't* get sent back to `/setup/site-setup/designSetup?siteSlug=SLUG.wordpress.com&theme=tsubaki` to activate the Woo theme, I go to the congrats page ahead. This isn't the case on staging - it works as expected.

It works as expected if I choose a theme and upgrade via /start

renatho committed:
Fix feature check to check based on the software set

renatho committed:
Merge branch 'trunk' of github.com:Automattic/wp-calypso into change/make-bundled-theme-generic

renatho commented:
> Doing some testing and after purchasing a business plan via /themes I don't get sent back to /setup/site-setup/designSetup?siteSlug=SLUG.wordpress.com&theme=tsubaki to activate the Woo theme, I go to the congrats page ahead. This isn't the case on staging - it works as expected.

@dsas, I couldn't reproduce the issue. Could you confirm if you followed some different steps compared to the following video? Or maybe you can also see the issue in my video and I misunderstood something?

https://github.com/Automattic/wp-calypso/assets/876340/28ce0857-26db-405b-b621-3fc4c2c9685e

(you will just see an extra refresh on the checkout page that I did for the credits)

And I didn't record, but after going through the WC setup wizard, I was redirected to the congrats page. Is this the page you saw directly after the checkout?

<img width="845" alt="Screenshot 2023-11-30 at 16 56 27" src="https://github.com/Automattic/wp-calypso/assets/876340/f05b014a-4e2a-47c3-abaa-1aa3555961e1">

dsas reviewed (changes requested):
Ah wait, I found the issue again. After the checkout I'm not being redirected to the theme, but to a 'Congrats' page.

https://github.com/Automattic/wp-calypso/assets/93301/e66b9af3-efa1-48df-9dd1-27bb7aa68f9a

It looks like this is a problem with the wpcalypso.wordpress.com environment. It happens when I use wpcalypso.wordpress.com directly, or when I test this PR with the calypso live link. It doesn't happen when I'm testing this PR locally, or if I'm just on WordPress.com.

renatho commented:
> It looks like this is a problem with the wpcalypso.wordpress.com environment. It happens when I use wpcalypso.wordpress.com directly, or when I test this PR with the calypso live link. It doesn't happen when I'm testing this PR locally, or if I'm just on WordPress.com.

To confirm, I tested it in a different PR not related to these changes: https://github.com/Automattic/wp-calypso/pull/84175, and I could also see the same issue there. So it's confirmed that it's some issue because of the test environment.

Thank you everyone for the reviews! I'll wait until Monday to deploy it to see if there's any other feedback, and to avoid the deployment on Friday. :)

a8ci18n commented:
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/9876186

Some locales (Hebrew, Japanese) have been temporarily machine-translated due to translator availability. All other translations are usually ready within a few days. Untranslated and machine-translated strings will be sent for translation next Monday and are expected to be completed by the following Friday.

Thank you @renatho for including a screenshot in the description! This is really helpful for our translators.

a8ci18n commented:
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/9876186).

a8ci18n commented:
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/9876186).

a8ci18n commented:
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/9876186).
